lib/screens/IngredientScreen.dart

```diff
@@ -296,445 +483,656 @@ class _IngredientsPageState extends State<IngredientsPage> {
-        crossAxisSpacing: 20,
-        mainAxisSpacing: 15,
-        children: ingredients);
-    return SingleChildScrollView(
-        child: Container(
-            width: MediaQuery.of(context).size.width,
-            height: bodyHeight,
-            decoration: const BoxDecoration(color: Colors.white),
-            child: Column(children: <Widget>[
-              Container(
-                  margin: const EdgeInsets.all(10),
-                  width: MediaQuery.of(context).size.width,
-                  decoration: const BoxDecoration(
-                    color: Colors.transparent,
-                  ),
-                  child: Text(sorted,
-                      style: const TextStyle(
-                        fontSize: ingredientInfoFontSize,
-                        color: Colors.black,
-                      ))),
-              Expanded(child: ingreds)
-            ])));
-  }
+    final res =
+        await Inventory.retrieveUserInventory(reverse, exDate, cat, alphabet);
+    bool success = false;
+    do {
+      if (res.statusCode == 200) {
+        var data = json.decode(res.body);
+        for (int i = 0; i < data.length; i += 2) {
+          for (var cats in data) {
+            List<IngredientData> ingredients = [];
+            for (var ingred in cats[1]) {
+              ingredients
+                  .add(IngredientData.create().inventoryIngredient(ingred));
+            }
+            inventory[cats[i]] = ingredients;
+          }
+        }
+        success = true;
+      } else {
+        int errorCode = await getDataRetrieveError(res.statusCode);
+        if (errorCode == 3) {
+          errorDialog(context);
+        }
+      }
+    } while (!success);
 
-  // TODO(18): integrate API into making ingredients
-  Widget IngredientTile() {
-    double tileHeight = 200;
+    return inventory;
+  }
 
-    bool expiresSoon = false;
-    bool expired = true;
+  Future<List<Widget>> BuildTiles() async {
+    List<Widget> toRet = [];
 
-    Container expires = Container(
-      height: tileHeight,
-      decoration: BoxDecoration(
-        borderRadius: const BorderRadius.all(Radius.circular(20)),
-        color: expiresSoon
-            ? Colors.red[200]
-            : (expired ? const Color(0xffFF0000) : Colors.white),
-      ),
-      child: Align(
-        alignment: FractionalOffset.bottomLeft,
-        child: Padding(
-          padding: const EdgeInsets.all(10),
-          child: Row(
-            children: <Widget>[
-              Expanded(
-                child: Text(
-                  expiresSoon ? 'Expires Soon!' : (expired ? 'Expired!' : ''),
-                  style: const TextStyle(
-                    color: Colors.white,
-                    fontSize: 16,
-                    fontWeight: FontWeight.w500,
-                  ),
-                ),
-              ),
-            ],
-          ),
+    for (var cat in userInventory.keys) {
+      if (userInventory[cat]!.length == 0) {
+        continue;
+      }
```

Comment:
Rather than checking this here and above, you could filter out empty categories when you're parsing the response JSON